product-mini/platforms/posix/main.c

```diff
@@ -643,14 +643,29 @@ main(int argc, char *argv[])
     }
 #endif
 
-    if (is_repl_mode)
+    ret = 0;
+    if (is_repl_mode) {
         app_instance_repl(wasm_module_inst);
-    else if (func_name)
-        app_instance_func(wasm_module_inst, func_name);
-    else
-        app_instance_main(wasm_module_inst);
+    }
+    else if (func_name) {
+        if (app_instance_func(wasm_module_inst, func_name)) {
```

Comment:
app_instance_func always return NULL, how about using `wasm_runtime_get_exception` to set the ret instead?

Reply:
oops, i forgot to add a few hunks. fixed.